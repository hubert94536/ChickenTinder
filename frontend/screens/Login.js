import React from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import { hideError, showError } from '../redux/Actions.js'
import { Image, ImageBackground, StyleSheet, Text, TouchableHighlight, View } from 'react-native'
import Alert from '../modals/Alert.js'
import { BlurView } from '@react-native-community/blur'
import colors from '../../styles/colors.js'
import Icon from 'react-native-vector-icons/FontAwesome'
import imgStyles from '../../styles/cardImage.js'
import loginService from '../apis/loginService.js'
import modalStyles from '../../styles/modalStyles.js'
import normalize from '../../styles/normalize.js'
import screenStyles from '../../styles/screenStyles.js'
import PropTypes from 'prop-types'

class Login extends React.Component {
  constructor() {
    super()
    this.state = {
      pressed: false,
      alert: false,
    }
  }

  async handleClick() {
    loginService
      .loginWithFacebook()
      .then((result) => {
        this.setState({ alert: false })
        this.props.navigation.replace(result)
      })
      .catch(() => {
        this.props.showError()
      })
  }

  cancelClick() {
    this.setState({ alert: false })
  }

  login() {
    this.setState({ alert: true })
  }

  render() {
    return (
      <ImageBackground source={require('../assets/backgrounds/Login.png')} style={styles.main}>
        <Image source={require('../assets/Icon_White.png')} style={styles.logo} />
        <View style={styles.bottom}> 
        <TouchableHighlight
          onShowUnderlay={() => this.setState({ phonePressed: true })}
          onHideUnderlay={() => this.setState({ phonePressed: false })}
          activeOpacity={1}
          underlayColor={'white'}
          onPress={() => this.props.navigation.replace('Phone')}
          style={[screenStyles.longButton, styles.phoneButton]}
        >
          <View style={screenStyles.contentContainer}>
            <Icon style={[imgStyles.icon, styles.buttonIcon]} name="phone" />
            <Text
              style={[
                screenStyles.longButtonText,
                this.state.phonePressed ? { color: colors.hex } : { color: 'white' },
              ]}
            >
              Login with Phone
            </Text>
          </View>
        </TouchableHighlight>
        <TouchableHighlight
          onShowUnderlay={() => this.setState({ pressed: true })}
          onHideUnderlay={() => this.setState({ pressed: false })}
          activeOpacity={1}
          underlayColor="white"
          onPress={() => this.login()}
          style={[screenStyles.longButton, styles.fbButton]}
        >
          <View style={[screenStyles.contentContainer]}>
            <Icon style={[imgStyles.icon, styles.buttonIcon]} name="facebook-official" />
            <Text
              style={[
                screenStyles.longButtonText,
                this.state.pressed ? { color: '#3b5998' } : { color: 'white' },
              ]}
            >
              Login with Facebook
            </Text>
          </View>
        </TouchableHighlight>

        <Text style={[screenStyles.textBook, styles.termsText]}>
          By clicking log in, you agree with our Terms and Conditions. Learn how we process your data in our Privacy Policy and Cookies Policy
        </Text>
        </View>
        {(this.state.alert || this.state.error) && (
          <BlurView
            blurType="dark"
            blurAmount={10}
            reducedTransparencyFallbackColor="white"
            style={modalStyles.blur}
          />
        )}
        {this.state.alert && (
          <Alert
            title='Open "Facebook?"'
            body="You will be directed to the Facebook app for account verification"
            twoButton
            buttonAff="Open"
            buttonNeg="Go Back"
            height="25%"
            press={() => this.handleClick()}
            cancel={() => this.cancelClick()}
          />
        )}
        {this.props.error && (
          <Alert
            title="Error, please try again"
            buttonAff="Close"
            height="20%"
            press={() => this.props.hideError()}
            cancel={() => this.props.hideError()}
          />
        )}
      </ImageBackground>
    )
  }
}

const mapStateToProps = (state) => {
  const { error } = state
  return { error }
}

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      showError,
      hideError,
    },
    dispatch,
  )

export default connect(mapStateToProps, mapDispatchToProps)(Login)

Login.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired,
    replace: PropTypes.func,
  }).isRequired,
  // error: PropTypes.bool,
  showError: PropTypes.func,
  hideError: PropTypes.func,
}

const styles = StyleSheet.create({
  main:{
    flex: 1,
    justifyContent:'space-evenly'
  },
  logo: {
    alignSelf: 'center',
    width: normalize(200),
    height: normalize(248),
    marginTop:'5%'
  },
  phoneButton: { borderColor: colors.hex, backgroundColor: colors.hex, marginTop: '5%' },
  buttonIcon: { fontSize: normalize(22), color: 'white', marginRight: '5%' },
  fbButton: { borderColor: '#3b5998', backgroundColor: '#3b5998', marginTop: '5%' },
  termsText: {
    alignSelf: 'center',
    marginHorizontal: '15%',
    marginTop: '5%',
    fontSize: normalize(13),
    textAlign: 'center',
    lineHeight: normalize(17),
  },
  bottom:{
    marginTop:'30%'
  }
})
