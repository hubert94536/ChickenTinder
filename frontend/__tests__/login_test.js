import 'react-native'
import React from 'react'
import Login from '../screens/login'
import renderer from 'react-test-renderer'

const facebookService = require('../apis/facebookService.js')
jest.mock('../apis/facebookService.js')

test('renders correctly', () => {
  const tree = renderer.create(<Login />).toJSON()
  expect(tree).toMatchSnapshot()
})
