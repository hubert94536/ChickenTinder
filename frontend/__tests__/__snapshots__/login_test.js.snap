// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders login page correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "flex": 1,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../frontend/assets/images/logo2.png",
      }
    }
    style={
      Object {
        "alignSelf": "center",
        "height": 248,
        "marginTop": "12%",
        "width": 200,
      }
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#F15763",
          "fontFamily": "CircularStd-Medium",
        },
        Object {
          "fontSize": 45,
          "textAlign": "center",
        },
        Object {
          "fontFamily": "CircularStd-Bold",
          "fontSize": 30,
          "fontWeight": "bold",
          "marginBottom": "10%",
          "marginTop": "2.5%",
        },
      ]
    }
  >
    Let's Get Chews-ing!
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignSelf": "center",
          "borderRadius": 30,
          "borderWidth": 2.5,
          "paddingHorizontal": 12,
          "paddingVertical": 5,
          "width": "70%",
        },
        Object {
          "backgroundColor": "#F15763",
          "borderColor": "#F15763",
          "marginTop": "7%",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "alignSelf": "center",
                "color": "#F15763",
                "fontFamily": "CircularStd-Medium",
                "fontSize": 35,
              },
              Object {
                "color": "white",
                "fontSize": 22,
                "marginRight": "5%",
              },
            ],
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Array [
            Object {
              "alignSelf": "center",
              "fontFamily": "CircularStd-Book",
              "fontSize": 18,
              "fontWeight": "normal",
            },
            Object {
              "color": "white",
            },
          ]
        }
      >
        Login with Phone
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignSelf": "center",
          "borderRadius": 30,
          "borderWidth": 2.5,
          "paddingHorizontal": 12,
          "paddingVertical": 5,
          "width": "70%",
        },
        Object {
          "backgroundColor": "#3b5998",
          "borderColor": "#3b5998",
          "marginTop": "7%",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "alignSelf": "center",
                "color": "#F15763",
                "fontFamily": "CircularStd-Medium",
                "fontSize": 35,
              },
              Object {
                "color": "white",
                "fontSize": 22,
                "marginRight": "5%",
              },
            ],
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Array [
            Object {
              "alignSelf": "center",
              "fontFamily": "CircularStd-Book",
              "fontSize": 18,
              "fontWeight": "normal",
            },
            Object {
              "color": "white",
            },
          ]
        }
      >
        Login with Facebook
      </Text>
    </View>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#F15763",
          "fontFamily": "CircularStd-Medium",
        },
        Object {
          "alignSelf": "center",
          "fontFamily": "CircularStd-Book",
          "fontSize": 13,
          "lineHeight": 17,
          "marginHorizontal": "15%",
          "marginTop": "7.5%",
          "textAlign": "center",
        },
      ]
    }
  >
    By clicking log in, you agree with our Terms and Conditions.
  </Text>
</View>
`;

exports[`renders login with Facebook alert 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "flex": 1,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../frontend/assets/images/logo2.png",
      }
    }
    style={
      Object {
        "alignSelf": "center",
        "height": 248,
        "marginTop": "12%",
        "width": 200,
      }
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#F15763",
          "fontFamily": "CircularStd-Medium",
        },
        Object {
          "fontSize": 45,
          "textAlign": "center",
        },
        Object {
          "fontFamily": "CircularStd-Bold",
          "fontSize": 30,
          "fontWeight": "bold",
          "marginBottom": "10%",
          "marginTop": "2.5%",
        },
      ]
    }
  >
    Let's Get Chews-ing!
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignSelf": "center",
          "borderRadius": 30,
          "borderWidth": 2.5,
          "paddingHorizontal": 12,
          "paddingVertical": 5,
          "width": "70%",
        },
        Object {
          "backgroundColor": "#F15763",
          "borderColor": "#F15763",
          "marginTop": "7%",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "alignSelf": "center",
                "color": "#F15763",
                "fontFamily": "CircularStd-Medium",
                "fontSize": 35,
              },
              Object {
                "color": "white",
                "fontSize": 22,
                "marginRight": "5%",
              },
            ],
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Array [
            Object {
              "alignSelf": "center",
              "fontFamily": "CircularStd-Book",
              "fontSize": 18,
              "fontWeight": "normal",
            },
            Object {
              "color": "white",
            },
          ]
        }
      >
        Login with Phone
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignSelf": "center",
          "borderRadius": 30,
          "borderWidth": 2.5,
          "paddingHorizontal": 12,
          "paddingVertical": 5,
          "width": "70%",
        },
        Object {
          "backgroundColor": "#3b5998",
          "borderColor": "#3b5998",
          "marginTop": "7%",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "alignSelf": "center",
                "color": "#F15763",
                "fontFamily": "CircularStd-Medium",
                "fontSize": 35,
              },
              Object {
                "color": "white",
                "fontSize": 22,
                "marginRight": "5%",
              },
            ],
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Array [
            Object {
              "alignSelf": "center",
              "fontFamily": "CircularStd-Book",
              "fontSize": 18,
              "fontWeight": "normal",
            },
            Object {
              "color": "white",
            },
          ]
        }
      >
        Login with Facebook
      </Text>
    </View>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#F15763",
          "fontFamily": "CircularStd-Medium",
        },
        Object {
          "alignSelf": "center",
          "fontFamily": "CircularStd-Book",
          "fontSize": 13,
          "lineHeight": 17,
          "marginHorizontal": "15%",
          "marginTop": "7.5%",
          "textAlign": "center",
        },
      ]
    }
  >
    By clicking log in, you agree with our Terms and Conditions.
  </Text>
  <View>
    <Text />
    <BlurView
      blurAmount={20}
      blurType="light"
      reducedTransparencyFallbackColor="white"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "bottom": 0,
            "elevation": 99,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 99,
          },
        ]
      }
    />
    <Modal
      animationType="none"
      hardwareAccelerated={false}
      transparent={true}
      visible={true}
    >
      <View
        style={
          Array [
            Object {
              "alignSelf": "center",
              "backgroundColor": "white",
              "borderRadius": 15,
              "elevation": 100,
              "justifyContent": "space-evenly",
              "margin": "50%",
              "width": "80%",
              "zIndex": 100,
            },
            Object {
              "height": "25%",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "flex-end",
            }
          }
        >
          <Text
            allowFontScaling={false}
            onPress={[Function]}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Object {
                  "color": "#F15763",
                  "fontFamily": "CircularStd-Medium",
                  "fontSize": 21,
                  "marginRight": "5%",
                  "marginTop": "5%",
                },
                Object {
                  "fontFamily": "anticon",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "space-evenly",
            }
          }
        >
          <View
            style={
              Object {
                "marginLeft": "10%",
                "marginRight": "10%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#F15763",
                  "fontFamily": "CircularStd-Medium",
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "marginBottom": "3%",
                }
              }
            >
              Open "Facebook?"
            </Text>
            <Text
              style={
                Object {
                  "fontFamily": "CircularStd-Medium",
                  "fontSize": 15,
                  "marginLeft": "2%",
                }
              }
            >
              You will be directed to the Facebook app for account verification
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "alignSelf": "center",
                    "borderColor": "#F15763",
                    "borderRadius": 60,
                    "borderWidth": 1.5,
                    "margin": "3%",
                    "width": "35%",
                  },
                  Object {
                    "marginBottom": "3%",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "CircularStd-Medium",
                      "fontSize": 20,
                      "paddingBottom": "5%",
                      "paddingTop": "5%",
                      "textAlign": "center",
                    },
                    Object {
                      "color": "#F15763",
                    },
                  ]
                }
              >
                Open
              </Text>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "alignSelf": "center",
                    "borderColor": "#F15763",
                    "borderRadius": 60,
                    "borderWidth": 1.5,
                    "margin": "3%",
                    "width": "35%",
                  },
                  Object {
                    "backgroundColor": "#F15763",
                    "marginBottom": "3%",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "CircularStd-Medium",
                      "fontSize": 20,
                      "paddingBottom": "5%",
                      "paddingTop": "5%",
                      "textAlign": "center",
                    },
                    Object {
                      "color": "white",
                    },
                  ]
                }
              >
                Go Back
              </Text>
            </View>
          </View>
        </View>
      </View>
    </Modal>
  </View>
</View>
`;
